lib/Wikia/tests/Util/SamplerProxyTest.php

```diff
@@ -0,0 +1,312 @@
+		$testAlternateCallable = [ $this->alternateMock, 'alternateMethod' ];
+		$resultsCallable = [ $this->alternateMock, 'compareResults' ];
+
+		$testArg1 = 1;
+		$testArg2 = 'two';
+		$testArg3 = array( 4, 5, 6 );
+		$originalTestResult = 'original';
+		$alternateTestResult = 'alternate';
+
+		$builder = SamplerProxy::createBuilder();
+		$samplerProxy =
+			$builder->enableShadowingVariableName( $testShadowVariableName )
+				->methodSamplingRateVariableName( $testMethodSamplingRateVariableName )
+				->originalCallable( $testOriginalCallable )
+				->alternateCallable( $testAlternateCallable )
+				->resultsCallable( $resultsCallable )
+				->build();
+
+		$this->originalMock->expects( $this->once() )
+			->method( 'methodToSample' )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $originalTestResult );
+
+		$this->alternateMock->expects( $this->once() )
+			->method( 'alternateMethod' )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $alternateTestResult );
+
+		$this->alternateMock->expects( $this->once() )
+			->method( 'compareResults' )
+			->with( $originalTestResult, $alternateTestResult )
+			->willReturn( $alternateTestResult );
+
+		$result = $samplerProxy->methodToSample( $testArg1, $testArg2, $testArg3 );
+
+		$this->assertEquals( $alternateTestResult, $result );
+	}
+
+	function testExceptionFallbackFromShadow() {
+		$testShadowVariableName = 'wgEnableShadowing';
+		$GLOBALS[$testShadowVariableName] = 0;
+		$testMethodSamplingRateVariableName = 'wgMethodSamplingRate';
+		$GLOBALS[$testMethodSamplingRateVariableName] = 100;
+
+		$testOriginalCallable = [ $this->originalMock, 'methodToSample' ];
+		$testAlternateCallable = [ $this->alternateMock, 'alternateMethod' ];
+
+		$testArg1 = 1;
+		$testArg2 = 'two';
+		$testArg3 = array( 4, 5, 6 );
+		$originalTestResult = 'original';
+
+		$builder = SamplerProxy::createBuilder();
+		$samplerProxy =
+			$builder->enableShadowingVariableName( $testShadowVariableName )
+				->methodSamplingRateVariableName( $testMethodSamplingRateVariableName )
+				->originalCallable( $testOriginalCallable )
+				->alternateCallable( $testAlternateCallable )
+				->build();
+
+		$this->alternateMock->expects( $this->once() )
+			->method( 'alternateMethod' )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willThrowException( new \Exception( 'test exception' ) );
+
+		$this->originalMock->expects( $this->once() )
+			->method( 'methodToSample' )
+			->with( $testArg1, $testArg2, $testArg3 )
+			->willReturn( $originalTestResult );
+
+
+		$result = $samplerProxy->methodToSample( $testArg1, $testArg2, $testArg3 );
+
+		$this->assertEquals( $originalTestResult, $result );
+	}
+
+}
+
+class OriginalPopo {
+
+	function methodToSample( $arg1, $arg2, $arg3 ) {
+		return 'original';
+	}
+
+	function methodNotToSample( $arg1, $arg2 ) {
+
+	}
+}
+
+class AlternatePopo {
+	function alternateMethod( $arg1, $arg2, $arg3 ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameters "$arg1, $arg2, $arg3". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)